src/include/storage/tuple_access_strategy.h

```diff
@@ -94,6 +97,19 @@ class TupleAccessStrategy {
    */
   explicit TupleAccessStrategy(BlockLayout layout);
 
+  /**
+   * Initializes a new block to conform to the layout given. This will write the
+   * headers and divide up the blocks into mini blocks(each mini block contains
+   * a column). The raw block needs to be 0-initialized (by default when given out
+   * from a block store), otherwise it will cause undefined behavior.
+   *
+   * @param data_table pointer to the DataTable to reference from this block
+   * @param raw pointer to the raw block to initialize
+   * @param layout_version the layout version of this block
+   */
+  void InitializeRawBlockForDataTable(storage::DataTable *data_table, RawBlock *raw,
+                                      layout_version_t layout_version) const;
```

Comment:
Why are we creating a new function rather than modifying/replacing the existing function.  Is there a nuance here where some blocks need to be initialized with a DataTable pointer and others do not?